modules/gateway/upnp.go

```diff
@@ -94,6 +104,15 @@ func (g *Gateway) threadedForwardPort(port string) {
 		return
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		select {
+		case <-g.threads.StopChan():
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
 	d, err := upnp.Discover()
```

Comment:
Shouldn't this be `d, err := upnp.DiscoverCtx(ctx)`? Seems like you only changed it the first time and forgot the other 5.

Reply:
derp, good catch